analyze/analyze.rs

```diff
@@ -601,8 +600,9 @@ impl traits::Emit for Paths {
             let item = &items[id];
             let size = item.size();
             let size_percent = (size as f64) / (items.size() as f64) * 100.0;
-            let mut path = String::with_capacity(item.name().len());
-            path.push_str(item.name());
+            let mut callers = items.predecessors(id).into_iter().map(|i| items[i].name()).collect::<Vec<&str>>();
```

Comment:
This isn't quite right either.

If we have

```
fn a() {}
fn b() { a() }
fn c() { a() }
fn d() { c() }
```

Then `a`'s predecessors will be `[b, c]`. But the paths to `a` are `[ "a <- b", "a <- c <- d" ]`. The predecessors just give us one depth of incoming edges, not the paths, which we construct recursively by processing the edges.

As written, this will emit `b -> c -> a` for the csv paths for `a`, but that isn't an actual path to `a`. It should emit two paths:

```
b -> a
d -> c -> a
```

The most straightforward way to fix this is to do a similar thing as `recursive_callers` does in `emit_text` and `emit_json` to use the predecessors to walk the paths.

Long term, every single `emit_blah` shouldn't have to redo this walk, and I have been meaning to fix this by doing the walking in the `paths` function that creates the `Paths` object and just storing the results in some sort of tree structure.